ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java

```diff
@@ -189,7 +189,22 @@ public void onResponse(Response response) throws IOException {
               callback.invoke(0, null, e.getMessage());
               return;
             }
-            callback.invoke(response.code(), null, responseBody);
+
+            WritableMap responseHeaders = Arguments.createMap();
+            Headers headers = response.headers();
+            for (int i = 0; i < headers.size(); i++) {
+              // multiple values for the same header
+              if (responseHeaders.hasKey(headers.name(i))) {
+                responseHeaders.putString(
+                  headers.name(i),
+                  responseHeaders.getString(headers.name(i) + "," + headers.value(i))
```

Comment:
Yes. HTTP RFC2616 available says: 
Multiple message-header fields with the same field-name MAY be present in a message if and only if the entire field-value for that header field is defined as a comma-separated list [i.e., #(values)]. It MUST be possible to combine the multiple header fields into one "field-name: field-value" pair, without changing the semantics of the message, by appending each subsequent field-value to the first, each separated by a comma. The order in which header fields with the same field-name are received is therefore significant to the interpretation of the combined field value, and thus a proxy MUST NOT change the order of these field values when a message is forwarded.